backpack-compose/src/main/kotlin/net/skyscanner/backpack/compose/bottomsheet/internal/BpkModalBottomSheetImpl.kt

```diff
@@ -0,0 +1,198 @@
+import net.skyscanner.backpack.compose.theme.BpkTheme
+import net.skyscanner.backpack.compose.tokens.BpkBorderRadius
+import net.skyscanner.backpack.compose.tokens.BpkElevation
+import net.skyscanner.backpack.compose.tokens.BpkSpacing
+import net.skyscanner.backpack.compose.tokens.NativeAndroidClose
+
+/**
+ * Backpack for Android - Skyscanner's Design System
+ *
+ * Copyright 2018 Skyscanner Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+@OptIn(ExperimentalMaterial3Api::class)
+@Composable
+internal fun BpkModalBottomSheetImpl(
+    onDismissRequest: () -> Unit,
+    state: BpkModalBottomSheetState,
+    dragHandleStyle: BpkDragHandleStyle,
+    title: String?,
+    isClosable: BpkModalBottomSheetCloseAction,
+    content: @Composable ColumnScope.() -> Unit,
+    modifier: Modifier = Modifier,
+) {
+    ModalBottomSheet(
+        sheetState = state.delegate,
+        content = {
+            BottomSheetContent(dragHandleStyle = dragHandleStyle, content = {
+                BpkModalBottomSheetContent(
+                    dragHandleStyle = dragHandleStyle,
+                    title = title,
+                    isClosable = isClosable,
+                    state = state,
+                    onDismissRequest = onDismissRequest,
+                    content = content,
+                )
+            })
+        },
+        modifier = modifier,
+        dragHandle = { if (dragHandleStyle == BpkDragHandleStyle.Default) BpkBottomSheetHandle() },
+        shape = RoundedCornerShape(topStart = BpkBorderRadius.Lg, topEnd = BpkBorderRadius.Lg),
+        containerColor = BpkTheme.colors.surfaceElevated,
+        contentColor = BpkTheme.colors.textPrimary,
+        scrimColor = BpkTheme.colors.scrim,
+        tonalElevation = BpkElevation.Lg,
+        onDismissRequest = onDismissRequest,
+    )
+}
+
+@Composable
+fun ColumnScope.BpkModalBottomSheetContent(
+    dragHandleStyle: BpkDragHandleStyle,
+    title: String?,
+    isClosable: BpkModalBottomSheetCloseAction,
+    state: BpkModalBottomSheetState,
+    onDismissRequest: () -> Unit,
+    content: @Composable ColumnScope.() -> Unit,
+    modifier: Modifier = Modifier,
+) {
+    if (isClosable is BpkModalBottomSheetCloseAction.Default || !title.isNullOrEmpty()) {
+        when (dragHandleStyle) {
+            BpkDragHandleStyle.Default -> {
+                Column(modifier = modifier,
+                    verticalArrangement = Arrangement.spacedBy(BpkSpacing.Lg),
```

Comment:
as we're exposing the column scope to the content this will mean every other item within the content will have a larger spacing - this is probably a case where we want a spacer. also not sure the spacing is right? i don't think there's any spacing between the header + content enforced